.github/workflows/update-helm-charts-index.yml

```diff
@@ -0,0 +1,69 @@
+name: update-helm-charts-index
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+jobs:
+  update-helm-charts-index:
+    outputs:
+      go-version: ${{ steps.get-go-version.outputs.version }}
+      success-message: ${{ steps.prepare-slack-notification.outputs.success-message }}
+      failure-message: ${{ steps.prepare-slack-notification.outputs.failure-message }}
+      success-boolean: ${{ steps.prepare-slack-notification.outputs.success-boolean }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: Determine Go version
+        id: get-go-version
+        run: |-
+          echo "Building with Go $(cat ".go-version")"
+          echo "go-version=$(cat ".go-version")" >> "$GITHUB_OUTPUT"
+      - name: Setup Go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: ${{ steps.get-go-version.outputs.go-version }}
+      - name: verify Chart version matches tag version
+        run: |-
+          git_tag=$(echo "${{ github.ref_name }}" | sed 's/v//g')
+          chart_tag=$(yq r Chart.yaml version)
+          if [ "${git_tag}" != "${chart_tag}" ]; then
+            echo "chart version (${chart_tag}) did not match git version (${git_tag})"
+            exit 1
+          fi
+      - name: update helm-charts index
+        id: update
+        env:
+          GH_TOKEN: ${{ secrets.HELM_CHARTS_GITHUB_TOKEN }}
+        run: |-
+          gh workflow run publish-charts.yml \
+            --repo hashicorp/helm-charts \
+            --ref main \
+            -f SOURCE_TAG="${{ github.ref_name }}" \
+            -f SOURCE_REPO="${{ github.repository }}"
+      - name: Prepare Slack Notification
+        id: prepare-slack-notification
+        run: |
+          case "${{ steps.update.conclusion }}" in
+            failure)
+              echo "success-boolean=false" >> $GITHUB_OUTPUT
+              ;;
+            success)
+              echo "success-boolean=true" >> $GITHUB_OUTPUT
+              ;;
+          esac
+          SUCCESS_MESSAGE=":tada: The waypoint-helm charts index update succeeded."
+          FAILURE_MESSAGE=":red_circle: The waypoint-helm charts index update failed."
+          echo "success-message=${SUCCESS_MESSAGE}" >> $GITHUB_OUTPUT
+          echo "failure-message=${FAILURE_MESSAGE}" >> $GITHUB_OUTPUT
+      - name: Send Slack Notification
+        # note: this will skip Slack notification if the job is skipped or cancelled.
+        if: ${{ (success() || failure()) }}
+        # FIXME: use new composite actions and ensure all inputs / name align
+        uses: hashicorp/actions-send-slack-status@v1
```

Comment:
This seems to be called `https://github.com/hashicorp/action-slack-status` so we should pick one and be consistent, either seems fine to me 😄